### DISCUSSION TITLE
calling an API in onSubmit doesn't gets called for first time,but gets called for second time

### DISCUSSION THREAD

**Discussion by waqarmr22** [react-hook-form/react-hook-form#5917] [Q&A]
**Describe the bug**
I am calling one APi in onSubmit().The API gets called for the second time and not for the first time.The 'data' fields are populating but when calling any API the fields come as empty

 const onSubmit = (data) => {
    console.log('data', data);
    setFirstName(data.firstName);

    setLastName(data.lastName);
    setEmail(data.email);
    setPhone(data.phone);
    setPassword(data.password);
APICall()  
};

<Controller
                  control={control}
                  rules={{
                    required: true,
                  }}
                  render={({ field: { onChange, onBlur, value } }) => (
                    <Input
                      style={styles.input}
                      onBlur={onBlur}
                      onChangeText={onChange}
                      value={value}
                      placeholder="First Name"
                    />
                  )}
                  name="firstName"
                  defaultValue=""
                />

            <Button buttonName="Submit" onPress={handleSubmit(onSubmit)} />

**Comment by jorisre** [comment]
Please follow https://github.com/react-hook-form/react-hook-form/issues/2848 🙏🏻
